### ISSUE TITLE
How to create .lst file for this project

### ISSUE THREAD

**Issue by prashantg445** [deepinsight/insightface#345]
Hi @nttstar  @yingfeng  @yuzhichang  @zhangxu19830126  @kernel8liang 
I want to use my own dataset for finetuning the pre-trained age-gender model(gamodel-r50). And I am creating a directory 'all_ages' in the same directory in which face2rec.py(src/data/) is. Then I have 100 sub-directories in this directory data corresponding to 0-99 ages, which further contains images in it accordingly. But face2rec.py function doesn't contain code or make_list function (even if I pass --list True to parser). The other option I am left with is to use im2rec.py from mxnet

But the format of .lst file that im2rec.py generates doesn't matches with the format that parse_lst_line (face_preprocess.py) function processes.

```
def parse_lst_line(line):
  vec = line.strip().split("\t")
  assert len(vec)>=3
  aligned = int(vec[0])
  image_path = vec[1]
  label = int(vec[2])
  bbox = None
  landmark = None
  #print(vec)
  if len(vec)>3:
    bbox = np.zeros( (4,), dtype=np.int32)
    for i in xrange(3,7):
      bbox[i-3] = int(float(vec[i]))
    landmark = None
    if len(vec)>7:
      _l = []
      for i in xrange(7,17):
        _l.append(float(vec[i]))
      landmark = np.array(_l).reshape( (2,5) ).T
  #print(aligned)
  return image_path, label, bbox, landmark, aligned
```
It seems like aligned accepts a number(probably 0 or 1). Hence, I guess the format of a line in .lst file will be:

`aligned    img_path     age    bbox_x1     bbox_y1    bbox_x2    bbox_y2    p1    p2    p3    p4    p5    p6    p7    p8    p9    p10 `
Am I correct? If not, then please provide the right way to do it.

Also, after creating .lst file, should I use face2rec.py as it is to create .rec file

**Comment by nttstar** [comment]
Right. It is better to align the face images at the beginning. Then the lines should be 
``1 image_path age`` (use TAB for blank)

**Comment by prashantg445** [comment]
I have some questions:

1. So, to include a image in lst file, I have to always put 1 in my first column, i.e., aligned=1? 
2. Age-gender model has label-width 2(predicting 2 labels- age and gender). So, to use that what my labels should be? **[gender,age]** or **gender,age** or **gender \t age**
3. What exactly do you mean by align the face images at the beginning. Use aligned images as my dataset or make lst file like
`1 img_path label bbox_x1 bbox_y1 bbox_x2 bbox_y2 p1 p2 p3 p4 p5 p6 p7 p8 p9 p10`
4. How my data should look like? Should it be all images in one folder with names like
`index_name_age_gender.jpg`
example
`10001_GoldieHawn_23_f.jpg`

Please reply for these 4 questions. Also, I am thinking to contribute in your project. Please assign me the tasks once I get proper understanding of it.

**Comment by nttstar** [comment]
1. Yes, the first column indicates whether face images in this lst were already aligned, it should be 0 or 1.
2. Reply to you later.
3. Bounding box and landmark locations are not required if you already aligned them.
4. Now it should be absolute path.

The contribution is very welcome. You can contact me via email~

**Comment by shangleyi** [comment]
If your images are aligned and cropped to the right size (in this case 112x112), you can refer to glint2lst.py directly and change the script to generate a lst file. The lst file I got from it has the format 1	ADDRESS 	LABEL. If your images are not aligned, align_megaface.py can help with the alignment and it does generate a lst file with the right format for you.

**Comment by zhangxiaopang88** [comment]
您好.之前没用mxnet.想问一下在生成自己的数据的时候，label是不是一定要从0开始，如果不是从0开始，设置face2rec2.py文件中recursive为true可以吗

**Comment by zhangxiaopang88** [comment]
读的.lst里的label

**Comment by shangleyi** [comment]
@zhangxiaopang88 
我是处理的时候把label设为从0开始，没有尝试设置recursive，不知道是否可行

**Comment by xizi** [comment]
@prashantg445 @nttstar Have you solved how to use labels? [gender,age] or gender,age or gender \t age

**Comment by SongQingChe** [comment]
I still dont know what format the labels are...

**Comment by ChenyangWang1** [comment]
> I still dont know what format the labels are...

use '/t'

**Comment by SongQingChe** [comment]
> > I still dont know what format the labels are...
> 
> use '/t'

Im so appreciate to your reply, my confusion is the label format because it has two kinds of labels, In the .lst file, does label obeys the format of  'gender,age' or something else? thank you so much for your further reply.

**Comment by SongQingChe** [comment]
>     1. Yes, the first column indicates whether face images in this lst were already aligned, it should be 0 or 1.
> 
>     2. Reply to you later.
> 
>     3. Bounding box and landmark locations are not required if you already aligned them.
> 
>     4. Now it should be absolute path.
> 
> 
> The contribution is very welcome. You can contact me via email~

I want to finetune this model, could you please tell me what the label format is in .lst file? is it the [gender age]? Im looking forward to seeing your reply, thanks!

**Comment by eeric** [comment]
introduction
https://github.com/eeric/Face_recognition_cnn/blob/main/code/insightface/README.md

**Comment by RamatovInomjon** [comment]
> 1. Yes, the first column indicates whether face images in this lst were already aligned, it should be 0 or 1.
> 2. Reply to you later.
> 3. Bounding box and landmark locations are not required if you already aligned them.
> 4. Now it should be absolute path.
> 
> The contribution is very welcome. You can contact me via email~
id \t gender\t age \t image_path
1	0	18	image___10000109_0.jpg
2	0	20	image___10000109_1.jpg
3	0	19	image___10000109_2.jpg
4	0	19	image___10000109_3.jpg
5	1	31	image___10000109_4.jpg
6	0	22	image___10000109_5.jpg
.lst file format


https://github.com/apache/incubator-mxnet/blob/master/tools/im2rec.py -> this file is for converting image to rec format
 you can run : python img2rec.py   prefix  img_dir/  --resize=112

to start tran : CUDA_VISIBLE_DEVICES=0 python train.py --data-dir path_to_train.rec_file/  --network 'm'

in training 2 options: m for mobilenet, and r for resnet models

**Comment by RamatovInomjon** [comment]
Hello guys @nttstar @xizi @eeric @prashantg445 @zhangxiaopang88 . Thanks! This repo is awsome! 
pretrained models for age gender trained using mobilenet or is there other backbone used? I got "Incompatible attr in node  at 0-th output: expected [256], got [1024]" error while using pretrained weights to start training
